src/DataTables/doc/index.rst

```diff
@@ -0,0 +1,146 @@
+
+.. caution::
+
+    Before you start, make sure you have `StimulusBundle configured in your app`_.
+
+Install the bundle using Composer and Symfony Flex:
+
+.. code-block:: terminal
+
+    $ composer require symfony/ux-datatables
+
+If you're using WebpackEncore, install your assets and restart Encore (not
+needed if you're using AssetMapper):
+
+.. code-block:: terminal
+
+    $ npm install --force
+    $ npm run watch
+
+    # or use yarn
+    $ yarn install --force
+    $ yarn watch
+
+Usage
+-----
+
+To use Symfony UX DataTables, inject the ``DataTableBuilderInterface`` service
+and create tables in PHP::
+
+    // ...
+    use Symfony\UX\DataTables\Builder\DataTableBuilderInterface;
+    use Symfony\UX\DataTables\Model\DataTable;
+
+    class HomeController extends AbstractController
+    {
+        #[Route('/', name: 'app_homepage')]
+        public function index(DataTableBuilderInterface $tableBuilder): Response
+        {
+            $table= $tableBuilder->('usersTable');
+
+            $table->setData([
+                'columns' => ['First name', 'Last name'],
+                'data' => [
+                    ['John', 'Doe'],
+                    ['Jane', 'Smith'],
+                ],
+            ]);
+
+            $table->setOptions([
+                'order' => [
+                    ['idx': 1, 'dir': 'asc']
+                ],
+            ]);
+
+            return $this->render('home/index.html.twig', [
+                'table' => $table,
+            ]);
+        }
+    }
+
+All options and data are provided as-is to DataTables. You can read
+`DataTables documentation`_ to discover them all.
+
+Once created in PHP, a table can be displayed using Twig:
+
+.. code-block:: html+twig
+
+    {{ render_datatable(table) }}
+
+    {# You can pass HTML attributes as a second argument to add them on the <table> tag #}
+    {{ render_datatable(table, {'class': 'my-table'}) }}
+
+Extend the default behavior
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Symfony UX DataTables allows you to extend its default behavior using a
+custom Stimulus controller:
+
+.. code-block:: javascript
+
+    // mytable_controller.js
+
+    import { Controller } from '@hotwired/stimulus';
+
+    export default class extends Controller {
+        connect() {
+            this.element.addEventListener('datatables:pre-connect', this._onPreConnect);
+            this.element.addEventListener('datatables:connect', this._onConnect);
+        }
+
+        disconnect() {
+            // You should always remove listeners when the controller is disconnected to avoid side effects
+            this.element.removeEventListener('datatables:pre-connect', this._onPreConnect);
+            this.element.removeEventListener('datatables:connect', this._onConnect);
+        }
+
+        _onPreConnect(event) {
+            // The table is not yet created
+            // You can access the config that will be passed to "new DataTable()"
+            console.log(event.detail.config);
```

Comment:
Looking at the Controller, it looks like this should be `event.detail.table` instead

Reply:
No, for the pre-connect event the config is being dispatched : 

```js
this.dispatchEvent('pre-connect', {
    config: payload,
});
```

But for the connect event, it's the table :

```js
this.dispatchEvent('connect', { table: this.table });
```